src/psyGen.py

```diff
@@ -1257,7 +1426,29 @@ class GlobalSum(Node):
     manipulated in, a schedule. '''
     def __init__(self, scalar, parent=None):
         Node.__init__(self, children=[], parent=parent)
-        self._scalar = scalar
+        import copy
+        self._scalar = copy.copy(scalar)
+        if scalar:
+            # update scalar values appropriately
+            # HACK:TODO: update mapping to readwrite when it is supported
```

Comment:
Yes we do zero sums. However, this is the global sum object which is a different beast. This takes partial sums from each mpi process (which indeed are each zero'ed), sums them up then updates the partial sums on each mpi process with the summed value.